src/borg/archive.py

```diff
@@ -468,6 +470,8 @@ def save(self, name=None, comment=None, timestamp=None, additional_metadata=None
         data = self.key.pack_and_authenticate_metadata(metadata.as_dict(), context=b'archive')
         self.id = self.key.id_hash(data)
         self.cache.add_chunk(self.id, Chunk(data), self.stats)
+        self.repository.async_wait()
```

Comment:
Ok, this is getting confusing with the non real time interface in github and edits.

> I think each time the borg server sends an error it will also free space in the other direction so that a command can be send

I was referring to the (byte stream level) windowing that the lower layers (pipe buffers, ssh, tcp) already implement.

> I believe so, with the current state. As we always immediately check for async_responses after sending an async request it should be virtually identical to what the previous code did 

I think the shutdown paths are not covered. Not sure if the new or the old behavior is better though.

Old: With 2 or more async errors:
* random operation is aborted with first error.
* `__leave__` of the repository context manager is called.
* issues a rollback
* is aborted with next async error
* finally does a close which closes all remote file descriptors and waits for ssh to terminate

new: With 2 or more async error and not delete --force:
* async_response raises exception
* `__leave__` of the repository context manager is called.
* issues a rollback
* does not abort. fills RemoeRepository.async_responses while waiting for remote to send response to rollback
* finally does a close which closes all remote file descriptors and waits for ssh to terminate